src/box/memtx_space.c

```diff
@@ -1135,33 +1139,26 @@ memtx_build_on_replace_rollback(struct trigger *trigger, void *event)
 	(void)event;
 	struct index_build_on_rollback_data *data = trigger->data;
 	struct txn_stmt *stmt = data->stmt;
-	struct memtx_ddl_state *state = data->state;
-	/*
-	 * Old tuple's format is valid if it exists.
-	 */
-	assert(stmt != NULL);
-	assert(stmt->old_tuple == NULL ||
-	       memtx_tuple_validate(state->format, stmt->old_tuple) == 0);
+	struct index *index = index_weak_ref_get_index(&data->index_ref);
+	/* The index was altered - return. */
+	if (index == NULL)
+		return 0;
 
 	struct tuple *delete = NULL;
 	struct tuple *successor = NULL;
 	/*
 	 * Use DUP_REPLACE_OR_INSERT mode because if we tried to replace a tuple
 	 * with a duplicate at a unique index, this trigger would not be called.
 	 */
-	state->rc = index_replace(state->index, stmt->new_tuple,
-				  stmt->old_tuple, DUP_REPLACE_OR_INSERT,
-				  &delete, &successor);
-	if (state->rc != 0) {
-		diag_move(diag_get(), &state->diag);
-		return 0;
-	}
+	int rc = index_replace(index, stmt->new_tuple, stmt->old_tuple,
+			       DUP_REPLACE_OR_INSERT, &delete, &successor);
+	VERIFY(rc == 0);
```

Comment:
There's no point in using `VERIFY` instead of `assert` if the asserted statement has no side effects.

Anyway, why do you ignore the possible failure? Maybe, better return -1 in this case to make `txn_rollback_one_stmt()` panic?

Reply:
> There's no point in using VERIFY instead of assert if the asserted statement has no side effects.

I used it to avoid `assert(rc == 0); (void)rc;` construction.
Anyway, the whole comment is outdated since I significantly reworked the patch.